jobs/kpi-forecasting/kpi_forecasting/metric_hub.py

```diff
@@ -52,15 +59,39 @@ def __post_init__(self) -> None:
             "\n", "\n" + " " * 19
         )
 
+        # Add query snippets for segments
+        if self.segments:
+            segment_select_query = []
+            for alias, sql in self.segments.items():
+                segment_select_query.append(f"  {sql} AS {alias},")
+            self.segment_select_query = "\n    ".join(segment_select_query)
+            self.segment_groupby_query = "\n ,".join(self.segments.keys())
+
+        self.where = f"AND {self.where}" if self.where else ""
+
     def query(self) -> str:
         """Build a string to query the relevant metric values from Big Query."""
+        if self.segments:
+            return dedent(
+                f"""
+                SELECT {self.submission_date_column} AS submission_date,
+                    {self.metric.select_expr} AS value,
+                    {self.segment_select_query}
```

Comment:
Not sure if this is 'more pythonic' or cleaner but with some edits above you can combine this query string into 1 version (i.e., not have the if statement on line 74.

To do that I think you'd need to 

- initialize `segment_select_query` as "" above line 63 and added a leading comma to it inside the line 63 if statement
- do the same thing for `segment_groupby_query`

Reply:
Updated this, thanks!